sympolyc/src/expr.rs

```diff
@@ -0,0 +1,241 @@
+//! Multivariate polynomial
+//! See [these notes](https://hackmd.io/@dannywillems/SyHar7p5A) for more context.
+
+use std::{
+    collections::HashMap,
+    fmt::{Debug, Formatter, Result},
+    ops::{Add, Mul},
+};
+
+use ark_ff::PrimeField;
+use num_integer::binomial;
+use o1_utils::FieldHelpers;
+
+use crate::utils::{
+    compute_all_two_factors_decomposition, naive_prime_factors, PrimeNumberGenerator,
+};
+
+/// Represents a multivariate polynomial of degree `D` in `N` variables.
+/// The natural representation is the coefficients given in the monomial basis.
+pub struct MVPoly<F: PrimeField, const N: usize, const D: usize> {
+    coeff: Vec<F>,
+    // keeping track of the indices of the monomials that are normalized
+    // to avoid recomputing them
+    normalized_indices: Vec<usize>,
```

Comment:
Is this guy supposed to be with the polynomial ?
He only depends on N and D, we shouldn't carry him with every polynomial, nor compute him every time we create a polynomial

Reply:
> Is this guy supposed to be with the polynomial ?
> He only depends on N and D, we shouldn't carry him with every polynomial, nor compute him every time we create a polynomial

No, I want to move it out. Only for now.

Regarding the degree bound, answered below.